### ISSUE TITLE
NSLogger  1.7.0 and the kLoggerOption_LogToConsole option.

### ISSUE THREAD

**Issue by jgahide** [fpillet/NSLogger#209]
Hello,
we had an issue. We are sending the logs to the console and the NSLogger desktop app (logDesktop). At startup time, we have some logs in the xcode console but not in the logDesktop. After a bit of investigation, it seems the log queued before the connection to the logDesktop. I have found the code that causes our issue : 

LoggerClient.m line 949 :

```
        if (logToConsole)
        {
            pthread_mutex_lock(&logger->logQueueMutex);
            while (CFArrayGetCount(logger->logQueue))
            {
                LoggerLogToConsole((CFDataRef)CFArrayGetValueAtIndex(logger->logQueue, 0));
                CFArrayRemoveValueAtIndex(logger->logQueue, 0);
            }
            pthread_mutex_unlock(&logger->logQueueMutex);
            pthread_cond_broadcast(&logger->logQueueEmpty);
        }
```

These piece of code flush the log to the console only, if kLoggerOption_LogToConsole is on and the connection not yet established. I think this is a bug as the documentation does not mention that ("NSLogger can now both log to system console and send the logs over the wire"). Right now I have solved this by commenting the while statement. Can you fix this ?

thanks,

**Comment by jgahide** [comment]
Maybe I have been a little been too quick. If the application will never connect to an app, then the logs won't be displayed to the xcode console. But we are still experiencing our issue : "the logs are lost before the connection occurs.".

**Comment by fpillet** [comment]
I'm having the same issue, it's because of the change in the way console logs are being captured. I'll revert to the old code as this is affecting me and many others.

**Comment by jgahide** [comment]
Thanks for your time.
I will upgrade our library as soon it is fixed.

**Comment by nkanellopoulos** [comment]
@fpillet Hi Florent. I think we need to tweet the desktop viewer to add the option whether to display or not the console output. It can be very distracting at times. No way to turn it off now.

**Comment by fpillet** [comment]
@nkanellopoulos two things: 1. you can turn console logging off with log options, 2. you can filter these logs out by setting up a filter that excludes tags `stderr` and `stdout`

**Comment by fpillet** [comment]
Fixed the issue -- NSLogger 1.7.0 introduced a new way of capturing stdout/stderr (using dispatch sources) and this completely shut the mirroring that we could previously do to the connected console (Xcode...). I reverted the code to the previous version that was working well.

**Comment by fpillet** [comment]
This is in HEAD and will be in 1.8.0 that I'll release imminently.
